crates/daphne/src/protocol/aggregator.rs

```diff
@@ -421,24 +422,20 @@ impl DapTaskConfig {
         let mut consumed_reports = Vec::with_capacity(report_count_hint);
         let mut helper_shares = Vec::with_capacity(report_count_hint);
         {
-            let mut processed = HashSet::with_capacity(report_count_hint);
+            let mut processed = replay_protection
+                .enabled()
+                .then(|| HashSet::with_capacity(report_count_hint));
             let mut reports = pin!(reports);
             while let Some(report) = reports.next().await {
-                let enforce_replay_check = {
-                    #[cfg(feature = "report-generator")]
-                    {
-                        !crate::testing::report_generator::generator_replaying_reports()
+                if let Some(processed) = &mut processed {
+                    if processed.contains(&report.report_metadata.id) {
+                        return Err(fatal_error!(
```

Comment:
I'm not sure when this happened, but this shouldn't be a fatal error. Rather, it should be a DAP protocol abort triggered by a non-compliant leader. Would you mind fixing this here?

Reply:
Ohhh, yes you're right, if this is called by the Leader than it can only be a fatal error. (Arguably this is an issue with the protocol.)